__tests__/e2e/lib/global-setup.ts

```diff
@@ -12,40 +12,57 @@ import { SettingsWritingPage } from './pages/settings-writing-page';
     const browser = await chromium.launch();
-    const page = await browser.newPage();
+    const context = await browser.newContext( { recordVideo: { dir: artifactsDir } } );
+    const page = await context.newPage();
     const user = process.env.E2E_USER ? process.env.E2E_USER : 'vipgo';
     const pass = process.env.E2E_PASSWORD ? process.env.E2E_PASSWORD : 'password';
+    let success = true;
     page.setDefaultNavigationTimeout( timeout );
+    await context.tracing.start( { name: 'global-setup', screenshots: true, snapshots: true } );
 
-    // Log in to wp-admin
-    await page.goto( baseURL + '/wp-login.php', { waitUntil: 'networkidle' } );
-    const loginPage = new LoginPage( page );
-    await loginPage.login( user, pass );
+    try {
 
-    // Save API Nonce to Env Var
-    process.env.WP_E2E_NONCE = await page.evaluate( 'wpApiSettings.nonce' );
+        // Log in to wp-admin
+        await page.goto( baseURL + '/wp-login.php', { waitUntil: 'networkidle' } );
+        const loginPage = new LoginPage( page );
+        await loginPage.login( user, pass );
 
-    // Adjust Classic Editor plugin settings if is available
-    await page.goto( baseURL + '/wp-admin/options-writing.php', { waitUntil: 'networkidle' } );
+        // Save API Nonce to Env Var
+        process.env.WP_E2E_NONCE = await page.evaluate( 'wpApiSettings.nonce' );
 
-    const settingsWritingPage = new SettingsWritingPage( page );
-    if ( await settingsWritingPage.hasClassicEditor() ) {
-        await settingsWritingPage.allowBothEditors();
-    } else {
-        process.env.E2E_CLASSIC_TESTS = 'false';
-    }
+        // Adjust Classic Editor plugin settings if is available
+        await page.goto( baseURL + '/wp-admin/options-writing.php', { waitUntil: 'networkidle' } );
+
+        const settingsWritingPage = new SettingsWritingPage( page );
+        if ( await settingsWritingPage.hasClassicEditor() ) {
+            await settingsWritingPage.allowBothEditors();
+        } else {
+            process.env.E2E_CLASSIC_TESTS = 'false';
+        }
+
+        // Dismiss editor welcome
+        await page.goto( baseURL + '/wp-admin/post-new.php', { waitUntil: 'networkidle' } );
+        const editorPage = new EditorPage( page );
+        await editorPage.dismissWelcomeTour();
 
-    // Dismiss editor welcome
-    await page.goto( baseURL + '/wp-admin/post-new.php', { waitUntil: 'networkidle' } );
-    const editorPage = new EditorPage( page );
-    await editorPage.dismissWelcomeTour();
+    } catch( error ) {
+        // eslint-disable-next-line no-console
+        console.log( error );
+        success = false;
+    }
 
     // Save signed-in state
     await page.context().storageState( { path: storageState as string } );
 
-    await browser.close();
+    await context.tracing.stop( { path: artifactsDir + 'trace.zip' } );
+    await context.close();
+
+    if ( ! success ) {
+        throw new Error( 'Setup unsuccessful - Tests will not run' )
+    }
```

Comment:
@bsessions85 should we clean up the video and trace on success?

Reply:
@sjinks yeah, we probably should. I'll get that in there. Thanks!

Reply:
@sjinks  Got that added in. Thanks!